integrations/sbt-bloop/src/main/scala/bloop/integrations/sbt/SbtBloop.scala

```diff
@@ -1147,7 +1147,22 @@ object BloopDefaults {
       )
       .map(fail =>
         if (fail.nonEmpty) {
-          logger.error(s"Couldn't run bloopGenerate. Cause: ${fail.mkString("\n")}")
+          val messages = fail.flatMap { incomplete =>
+            incomplete.directCause.zip(incomplete.message)
+          }
+
+          if (messages.nonEmpty) {
+            logger.error(s"Couldn't run bloopGenerate. The issue might be caused by:")
+            messages.foreach {
+              case (t, message) =>
+                logger.error(message)
+                logger.trace(t)
+            }
+          } else {
+            logger.error(
+              s"Couldn't run bloopGenerate. The issue could be caused by failing to compile the project."
```

Comment:
Is it not supposed to not even try to compile the project?

Reply:
it should compile if there is source generation, so this might happen.

Reply:
I would argue that if there is source generation it should only generate the sources, not compile.

But in general we don't know what's in the build and a compilation could happen.